modules/react/text-input/stories/examples/Icons.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+
+import {xSmallIcon, mailIcon} from '@workday/canvas-system-icons-web';
+import {FormField} from '@workday/canvas-kit-react/form-field';
+import {TertiaryButton} from '@workday/canvas-kit-react/button';
+import {InputGroup} from '@workday/canvas-kit-react/text-input';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {dispatchInputEvent, styled} from '@workday/canvas-kit-react/common';
+
+// create a prop forwarding component for FormField to forward to
+const InputGroupFormFieldForwarder = (props: {}) => {
+  const inputRef = React.useRef(null);
+  const [value, setValue] = React.useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.currentTarget.value);
+  };
+
+  return (
+    <InputGroup width={280}>
+      <InputGroup.Start pointerEvents="none">
+        <SystemIcon icon={mailIcon} size="small" />
+      </InputGroup.Start>
+      <InputGroup.Start pointerEvents="none">
+        <SystemIcon icon={mailIcon} size="small" />
+      </InputGroup.Start>
+      <InputGroup.Input ref={inputRef} onChange={handleChange} {...props} />
+      <InputGroup.End pointerEvents="none">
+        <SystemIcon icon={mailIcon} size="small" />
+      </InputGroup.End>
+      <InputGroup.End>
+        <TertiaryButton
+          role="presentation"
+          icon={xSmallIcon}
+          size="small"
+          tabIndex={-1}
+          transition="opacity 300ms ease"
+          style={{opacity: value ? 1 : 1}} // Use style attribute to avoid the cost of Emotion's styling solution that causes the browser to throw away style cache. The difference can be significant for large amount of elements (could be a 80ms difference)
+          onMouseDown={event => {
+            event.preventDefault(); // prevent a focus change
+          }}
+          onClick={event => {
+            dispatchInputEvent(inputRef.current, '');
+          }}
+        />
+      </InputGroup.End>
+    </InputGroup>
+  );
+};
+
+const InputGroupFormFieldForwarder2 = (props: {}) => {
+  const inputRef = React.useRef(null);
+  const [value, setValue] = React.useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.currentTarget.value);
+  };
+
+  return (
+    <InputGroup width={280}>
+      <InputGroup.Start pointerEvents="none">
+        <SystemIcon icon={mailIcon} size="small" />
+      </InputGroup.Start>
+      <InputGroup.Input ref={inputRef} onChange={handleChange} {...props} />
+      <InputGroup.End>
+        <TertiaryButton
+          role="presentation"
+          icon={xSmallIcon}
+          size="small"
+          tabIndex={-1}
+          transition="opacity 300ms ease"
+          opacity={value ? 1 : 1}
```

Comment:
Did you decide not to use the `style` attr here? And I think this should be:
```tsx
opacity={value ? 1 : 0}
```

Reply:
This was my performance test story. I didn't mean to commit it. I tried a style prop, a style attribute, and a data attribute to test timing and demonstrate elements effected.

The browser does optimize over time. For example, the data attribute has 2 elements effected the first time, but caches the result so that the second time only 1 element is effected. The style prop is 14 effected the first time and 5 the second time. As the page changes, these optimizations are flushed, so it is still better to avoid Emotion for state-based style changes.